synapse/storage/state.py

```diff
@@ -578,318 +573,3 @@ def must_await_full_state(self, is_mine_id: Callable[[str], bool]) -> bool:
     types=frozendict({EventTypes.Member: frozenset()}), include_others=True
 )
 _NONE_STATE_FILTER = StateFilter(types=frozendict(), include_others=False)
```

Comment:
I wonder if we ought to move the `StateFilter` stuff too. Is it used anywhere other than to pass down to `StateGroupStorage` ?

Reply:
I'd be tempted to leave it where it is TBH, we tend to define types that are external to the `synapse.storage` module in top level files :shrug:

Reply:
I don't disagree, but I don't follow your reasoning.

> we tend to define types that are external to the synapse.storage module in top level files

... like what? And this is a type internal to the `synapse.storage` module, so how is it relevant?

Reply:
`synapse.storage.roommember` has a bunch of types in it for example?



> And this is a type internal to the `synapse.storage` module, so how is it relevant?

The `StateFilter` is used all over the code base to specify the filter. If you mean `_NONE_STATE_FILTER` and co, then they are used solely by the `StateFilter` class?

Reply:
oh right. by "external" you mean "part of the external api", sorry.

I'm not convinced that's a convention we actually follow, but I don't mind it staying here for now anyway.